### PR TITLE
Player

### PR DESCRIPTION
**Pull request by shloch**

### CODE REVIEW

**Review by inhaq — CHANGES_REQUESTED** [review]
Hi, good attempt to Tic Tac Toe and welcome to the OOP. In order to practice OOP, you must understand SOLID principle. It's not always about making app works but rather following practices which will help you when you'll need to do modification later. Lesser you'll depends, less you'll need to change.

 - There is no need to add `Winnermodule` from another file as it's just one array.
 - In `player.rb` you're using two big functions, slice it down to small functions. Like `user_input', `display_msg` etc
 - `gameclass.rb` has class which can be further sliced to functionality and display messages. Consider putting them to separate classes.
 - In execution file i.e. `game.rb` there is display messages as well as game functionality, kindly use it for execution only rather than putting both into one file.

I'm not able to run this project since you're using gems but `Gemfile` is missing from repo. Kindly add `Gemfile` and resubmit for review.

**Comment by shloch** [comment]
Thanks for the review....we proposed changes as demanded

File: lib/gameclass.rb
Comment (inhaq) [inline lib/gameclass.rb]: why use `@@` class variables?

**Review by inhaq — CHANGES_REQUESTED** [review]
Hi folks, nice try. But few of implementations from the last review wasn't done or done properly.
Update Gemfile to execute properly. Try to make your functions pure. It should do one job and do it very well. Try to not go over a few lines for every function.
Once updated, kindly submit for review.

File: lib/gameclass.rb
Comment (shloch) [reply lib/gameclass.rb]: changed

**Review by inhaq — CHANGES_REQUESTED** [review]
Hi guys, you're reaching the finishing line. Here are a few things to take care of:
 - Use specific names for files and functions. For example, two files are named as `game.rb` and `gameclass.rb`, try to be more specific about these classes and files.
 - Within `game.rb` and `player.rb` there is function `play`. Try to use a clearer name.
 - Both these functions should be sliced down to become pure function.

Once fixed, kindly submit for review.

**Review by inhaq — APPROVED** [review]
Hi folks, I understand that you're trying your best. But you're just changing minor things here and there and submit it again. This project is still not 100% and I would suggest doing these modifications before merging. I really want you guys to learn more and improve this project.
 - fix `rubocop` errors(it'll suggest slicing down the functions as well)
 - fix namings for the functions and classes. You can't name anything `gameClass` and then create player separately, it should be `player`, `board`, `messages` etc. And each class should responsible for their own functionalities. 
 - update `readme` as executable file isn't `game.rb` anymore.

Once fixed, merged the branch. No need to review from my side. Good luck 👍🏻

**Comment by shloch** [comment]
thanks for the reviews...we learned many things

**Pull request merged**
